docs/src/man/comparisons.md

```diff
@@ -1,77 +1,248 @@
+Julia functions do not skip `NaN`'s for safety reasons. If necessary, you can filter out
+the `NaN`'s before processing, for example, `mean(Iterators.filter(!isnan, x))`
+
+Pandas uses `NaN` for representing both missing data and the floating point "not a number" value.
+Julia defines a special value `missing` for representing missing data. DataFrames.jl respects
+general rules in Julia in propagating `missing` values by default. If necessary,
+the `skipmissing` function can be used to remove missing data.
+See the [Missing Data](@ref) section for more information.
+
+In addition, pandas keeps the original column name after applying a function.
+DataFrames.jl appends a suffix to the column name by default. To keep it simple, the
+examples above do not synchronize the column names between pandas and DataFrames.jl.
+
+### Grouping data and aggregation
+
+DataFrames.jl provides a `groupby` function to apply operations
+over each group independently. The result of `groupby` is a `GroupedDataFrame` object
+which may be processed using the `combine`, `transform`, or `select` functions.
+The following table illustrates some common grouping and aggregation usages.
+
+| Operation                       | pandas                                                                                | DataFrames.jl                                        |
+|---------------------------------|---------------------------------------------------------------------------------------|------------------------------------------------------|
+| Aggregate by groups             | `df.groupby('grp')['x'].mean()`                                                       | `combine(groupby(df, :grp), :x => mean)`             |
+| Rename column after aggregation | `df.groupby('grp')['x'].mean().rename("my_mean")`                                     | `combine(groupby(df, :grp), :x => mean => :my_mean)` |
+| Add aggregated data as column   | `df.join(df.groupby('grp')['x'].mean(), on='grp', rsuffix='_mean')`                   | `transform(groupby(df, :grp), :x => mean)`           |
+| ...and select output columns    | `df.join(df.groupby('grp')['x'].mean(), on='grp', rsuffix='_mean')[['grp','x_mean']]` | `select(groupby(df, :grp), :id, :x => mean)`         |
+
+Note that pandas returns a `Series` object for 1-dimensional result unless `reset_index` is called afterwards.
+The corresponding DataFrames.jl examples return an equivalent `DataFrame` object.
+Consider the first example:
+
+```python
+>>> df.groupby('grp')['x'].mean()
+grp
+1    4
+2    3
+Name: x, dtype: int64
+```
+
+For DataFrames.jl, it looks like this:
+
+```julia
+julia> combine(groupby(df, :grp), :x => mean)
+2×2 DataFrame
+│ Row │ grp   │ x_mean  │
+│     │ Int64 │ Float64 │
+├─────┼───────┼─────────┤
+│ 1   │ 1     │ 4.0     │
+│ 2   │ 2     │ 3.0     │
+```
+
+In DataFrames.jl, the `GroupedDataFrame` object that contains a dictionary for efficient key lookups.
```

Comment:
The fact that we use a `Dict` internally is an implementation detail that might change:
```suggestion
In DataFrames.jl, the `GroupedDataFrame` object supports an efficient key lookup.
```